mk/spksrc.common.mk

```diff
@@ -13,13 +13,22 @@ MSG = echo "===> "
 # Launch command in the working dir of the package source and the right environment
 RUN = cd $(WORK_DIR)/$(PKG_DIR) && env $(ENV)
 
-# Pip command
+# fallback by default to native/python*
 PIP ?= pip
+
+# Cross compiling must call "pip" directly to ensure using
+# $(WORK_DIR)/crossenv pip version instead from adjusted $PATH
+PIP_CROSS = $(shell . $(CROSSENV) && $(RUN) which pip)
+
+# System default pip outside from build environment
+PIP_SYSTEM = $(shell which pip)
```

Comment:
Actually this is a great question... In the docker image `pip` for python gets installed after resulting in an older version of `pip` instead of using the latest.  So to add to the confusion, here is what it looks like (and I have to rethink this for a bit):
```
$ pip3 --version
pip 21.3 from /usr/local/lib/python3.7/dist-packages/pip (python 3.7)
$ pip2 --version
pip 20.3.4 from /usr/local/lib/python2.7/dist-packages/pip (python 2.7)
$ pip --version
pip 20.3.4 from /usr/local/lib/python2.7/dist-packages/pip (python 2.7)
```

Reply:
Thanks for this. I think we can keep the old python2/pip in this PR but we should consider if we can move more to python3 and pip3 in the near future. (Hopefully without breaking things 🤞) @hgy59 might have more insight.

Reply:
Should be PIP_HOST imo

Reply:
Thanks, yea I think that works 👍